firedrake/mesh.py

```diff
@@ -976,7 +975,7 @@ def make_mesh_from_coordinates(coordinates):
             return mesh
 
     coordinates_fs = coordinates.function_space()
-    if not isinstance(coordinates_fs, functionspace.VectorFunctionSpace):
+    if coordinates_fs.rank != 1:
```

Comment:
Should one additionally assert that the `value_shape` has length 1?  The previous check did not do this.